### ISSUE TITLE
Problem with recursive [following issue #143]

### ISSUE THREAD

**Issue by xxXmatXxx** [gorakhargosh/watchdog#497]
``` python
#! /usr/bin/python

import os
import os.path
import sys
import time
import logging
from watchdog.observers import Observer
from watchdog.events import LoggingEventHandler

if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO,
                        format='%(asctime)s - %(message)s',
                        datefmt='%Y-%m-%d %H:%M:%S')
    path = sys.argv[1] if len(sys.argv) > 1 else '.'
    event_handler = LoggingEventHandler()
    observer = Observer()
    observer.schedule(event_handler, path, recursive=False)
    observer.start()

    if not os.path.isdir('watchdog_test'):
        os.mkdir('watchdog_test')
    open('watchdog_test/test.txt', 'w').close()

    try:
        while True:
            time.sleep(1)
    except KeyboardInterrupt:
        observer.stop()
    observer.join()
```

Executing the code above leads to the following output with v.0.6.0 (despite setting 'Recursive' to 'False'):

```
2013-12-11 16:43:05 - Created directory: /.../projects/watchdog/watchdog_test
2013-12-11 16:43:05 - Created file: /.../projects/watchdog/watchdog_test/test.txt
2013-12-11 16:43:05 - Modified file: /.../projects/watchdog/watchdog_test/test.txt
```

And to the following output when using the source version (which leads to correct behaviour):

```
2013-12-12 11:39:59 - Created directory: /.../projects/watchdog/watchdog_test
```

Therefore, the issue seems to be fixed and could be closed.

_Originally posted by @cpriebe in https://github.com/gorakhargosh/watchdog/issues/143#issuecomment-30405792_

**Comment by xxXmatXxx** [comment]
I still got report when creating folder or files in subdirectories on the watch folder,
I already tried the above solution but still got the issue,

I tried by installing from source file (downloaded from here https://pypi.org/project/watchdog/) and pip
My OS is Windows 10 and I use python 3.4

Thank You

**Comment by rrzaripov** [comment]
@xxXmatXxx can you test this changes https://github.com/gorakhargosh/watchdog/pull/454/commits/ce40f75a619f11dc305d5d8ebc2575fd0fc98bca if this resolve youre issue, I'm update my PR.

**Comment by xxXmatXxx** [comment]
Sorry, I'm not quite sure how to test the changes and where I should edit it on

**Comment by rrzaripov** [comment]
@xxXmatXxx for testing you can create separate virtual environment. Then activate it and install watchdog with patch from my branch using pip. And finally, start test application and make test actions as in #143

```
mkdir c:\tmp\test
cd c:\tmp\test
pip install virtualenv
virtualenv -p c:\path\to\python.exe venv
venv\Scripts\activate
pip install git+https://github.com/rrzaripov/watchdog.git@fix-recursive-issue-under-windows
python test.py
```

**Comment by BoboTiG** [comment]
Fixed with b8cf1b0.
